include/dali/core/mm/cu_vm.h

```diff
@@ -33,10 +33,16 @@ namespace cuvm {
  */
 struct CUAddressRange : std::pair<CUdeviceptr, size_t> {
   using std::pair<CUdeviceptr, size_t>::pair;
-  CUdeviceptr &ptr() noexcept { return first; }
-  CUdeviceptr  ptr() const noexcept { return first; }
-  size_t &size() noexcept { return second; }
-  size_t  size() const noexcept { return second; }
+  constexpr CUdeviceptr &ptr() noexcept { return first; }
+  constexpr CUdeviceptr  ptr() const noexcept { return first; }
+  constexpr size_t &size() noexcept { return second; }
+  constexpr size_t  size() const noexcept { return second; }
+
+  constexpr CUdeviceptr end() const noexcept { return first + second; }
```

Comment:
add begin as well(?)

Reply:
I didn't want them to look like iterators.